index.js

```diff
@@ -556,7 +557,9 @@ class KeyringController extends EventEmitter {
    * Attempts to initialize a new keyring from the provided
    * serialized payload.
    *
-   * On success, the resulting keyring instance.
+   * On success, returns the resulting keyring instance.
+   *
+   * ATTN: The caller must call _updateMemStoreKeyrings as necessary.
```

Comment:
It seems unreasonable to expect an consumer of this package to call an internal method. Also, this would be a breaking change, and this isn't the most visible way to call attention to this.

I'd suggest instead factoring out the bulk of `restoreKeyring` to an internal `_restoreKeyring` method that omits the `_updateMemStoreKeyrings` call. Then internally `_restoreKeyring` can be used, followed by the `_updateMemStoreKeyrings` call, but the publicly-facing `restoreKeyring` method can remain unchanged.

Reply:
Yeah, that was sloppy of me. Done in f90bd69